WireUI/Sources/WireUIFoundation/Main/Protocols/Coordinator/MainCoordinatorProtocol.swift

```diff
@@ -0,0 +1,26 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+public protocol MainCoordinatorProtocol: AnyObject {
+    func showConversationList<ConversationFilter>(conversationFilter: ConversationFilter?) async
+        where ConversationFilter: MainConversationFilterRepresentable
+    func showArchivedConversations() async
+    func showSelfProfile() async
+    func showSettings() async
+    func showNewConversation() async
+}
```

Comment:
related to my comment above, the MainCoordinator builder could have similar builder methods (buildSelfprofile, buildNewConversation..), so we have a consistency between the MainCoordinator protocol methods and MainCoordinator builder methods